compiler/src/dotty/tools/dotc/core/NullOpsDecorator.scala

```diff
@@ -59,65 +95,33 @@ object NullOpsDecorator {
     def maybeNullable(implicit ctx: Context): Type =
       if (ctx.explicitNulls) OrType(self, defn.NullType) else self
 
-    /** Normalizes unions so that all `Null`s (or aliases to `Null`) appear to the right of all other types.
-     *  In the process, it also flattens the type so that there are no nested unions at the top level.
-     *  e.g. `Null | (T1 | Null) | T2` => `T1 | T2 | Null | Null`
-     */
-    def normNullableUnion(implicit ctx: Context): Type = {
-      assert(ctx.explicitNulls)
-      def split(tp: Type, nonNull: List[Type], nll: List[Type]): (List[Type], List[Type]) = {
-        tp match {
-          case OrType(left, right) =>
-            // Recurse on the right first so we get the types in pre-order.
-            val (nonNull1, nll1) = split(right, nonNull, nll)
-            split(left, nonNull1, nll1)
-          case _ =>
-            if (tp.isNullType) (nonNull, tp :: nll) else (tp :: nonNull, nll)
-        }
-      }
-      val (nonNull, nll) = split(self, Nil, Nil)
-      val all = nonNull ++ nll
-      assert(all.nonEmpty)
-      all.tail.foldLeft(all.head)(OrType.apply)
-    }
-
-    /** Strips `Null` from type unions in this type.
-     *  @param onlyJavaNull if true, we delete only `JavaNull` and not vanilla `Null`.
-     */
-    private def stripNullImpl(onlyJavaNull: Boolean)(implicit ctx: Context): Type = {
-      assert(ctx.explicitNulls)
-      def strip(tp: Type, changed: Boolean): (Type, Boolean) = {
-        tp match {
-          case OrType(left, right) if right.isNullType =>
-            if (!onlyJavaNull || right.isJavaNullType) strip(left, changed = true)
-            else {
-              val (tp1, changed1) = strip(left, changed)
-              (OrType(tp1, right), changed1)
-            }
-          case _ => (tp, changed)
-        }
-      }
-      // If there are no `Null`s to strip off, try to keep the method a no-op
-      // by keeping track of whether the result has changed.
-      // Otherwise, we would widen and dealias as a side effect.
-      val (tp, diff) = strip(self.widenDealias.normNullableUnion, changed = false)
-      if (diff) tp else self
-    }
-
-    /** Syntactically strips the nullability from this type. If the normalized form (as per `normNullableUnion`)
-     *  of this type is `T1 | ... | Tk | ... | Tn`, and all types in the range `Tk ... Tn` are references to `Null`,
-     *  then return `T1 | ... | Tk-1`.
+    /** Syntactically strips the nullability from this type.
+     *  If the normalized form (as per `normNullableUnion`) of this type is `T1 | ... | Tn-1 | Tn`,
+     *  and `Tn-1` `Tn` are references to `Null`, then return `T1 | ... | Tn-2`.
```

Comment:
Will need to update the comment to reflect the fact that only the last component of the union will be Null or JavaNull.